In [1]:
import nltk

# Basics

In [2]:
text = "Python is an interpreted high-level programming language for general-purpose programming. Created by Guido van Rossum and first released in 1991."

In [3]:
# sentence tokenizer
sentences = nltk.sent_tokenize(text)
len(sentences)

2

In [4]:
# word tokenizer
words = nltk.word_tokenize(text)
len(words)

22

In [5]:
words[:5]

['Python', 'is', 'an', 'interpreted', 'high-level']

In [6]:
# frequency of words
wordfreq = nltk.FreqDist(words)
wordfreq.most_common(2)

[('programming', 2), ('.', 2)]

In [7]:
from nltk.book import *

*** Introductory Examples for the NLTK Book ***
Loading text1, ..., text9 and sent1, ..., sent9
Type the name of the text or sentence to view it.
Type: 'texts()' or 'sents()' to list the materials.
text1: Moby Dick by Herman Melville 1851
text2: Sense and Sensibility by Jane Austen 1811
text3: The Book of Genesis
text4: Inaugural Address Corpus
text5: Chat Corpus
text6: Monty Python and the Holy Grail
text7: Wall Street Journal
text8: Personals Corpus
text9: The Man Who Was Thursday by G . K . Chesterton 1908


**Searching for a pattern in a text**
- There are multiple ways of searching for a pattern in a text.
- The example shown below searches for words starting with tri, and ending with r.



In [8]:
text1.findall("<tri.*r>")

triangular; triangular; triangular; triangular


# Basic Tasks with Text

**Determining Total Word Count**

NLTK Book module consits of 9 text class. Type the name of the text or sentence to view it.
- text1: Moby Dick by Herman Melville 1851 
- text2: Sense and Sensibility by Jane Austen 1811 
- text3: The Book of Genesis 
- text4: Inaugural Address Corpus 
- text5: Chat Corpus 
- text6: Monty Python and the Holy Grail 
- text7: Wall Street Journal 
- text8: Personals Corpus 
- text9: The Man Who Was Thursday by G . K . Chesterton 1908

In [9]:
type(text1)

nltk.text.Text

In [10]:
n_words = len(text1)
n_words

260819

**Determining Unique Word Count**

In [11]:
# A unique number of words in text1 is determined using set and len methods.
n_unique_words = len(set(text1))
n_unique_words

19317

**Transforming Words**
- It is possible to apply a function to any number of words and transform them.
- Now let's transform every word of text1 to lowercase and determine unique words once again.

In [12]:
text1_lcw = [ word.lower() for word in set(text1) ]
n_unique_words_lc = len(set(text1_lcw))
n_unique_words_lc

17231